ts/src/bitmex.ts

```diff
@@ -923,10 +1073,7 @@ export default class bitmex extends Exchange {
         const type = this.parseLedgerEntryType (this.safeString (item, 'transactType'));
         const currencyId = this.safeString (item, 'currency');
         const code = this.safeCurrencyCode (currencyId, currency);
-        let amount = this.safeNumber (item, 'amount');
-        if (amount !== undefined) {
-            amount = amount / 100000000;
-        }
+        let amount = this.convertToRealAmount (code, this.safeString (item, 'amount'));
```

Comment:
few lines below, there is mutation of `amount` variable, so, from `convertToRealAmount` it might return `string` (because of `parseNumber`) so that down below code will crash as it assumes `amount` to be numeric.

Reply:
also, down below,  `feeCost`, `after` and `before` values should be also handled using `convertToRealAmount` & `Precise.` methods